File: ts/src/p3.ts
function getInput(): string {
  return `..##.......
#...#...#..
.#....#..#.
..#.#...#.#
.#...##..#.
..#.##.....
.#.#.#....#
.#........#
#.##...#...
#...##....#
.#..#...#.#`;
}



const map = getInput()
  .split("\n")
.map(x=> x.split("")); // .map(x=> x === "." ? MapTree.snow : MapTree.tree))

const colLen = map[0].length;
let result = 0;

map.forEach((row, i)=> {
  if(row[(i*3%colLen)] === "#"){
    result++;
  }
})

console.log("Tree Count", result);

File: rust/src/bin/p3.rs

fn get_input() -> &'static str {
"..##.......
#...#...#..
.#....#..#.
..#.#...#.#
.#...##..#.
..#.##.....
.#.#.#....#
.#........#
#.##...#...
#...##....#
.#..#...#.#"
}

fn main() {
    let result = get_input()
        .lines()
        .enumerate()
        .flat_map(|(idx, line)| line.chars().nth(idx * 3 % line.len()))
        .filter(|&c| c == '#')
        .count();

    println!("Tree: {}", result);
}
